### DISCUSSION TITLE
Suppress runtime error when glob doesn't match any files?

### DISCUSSION THREAD

**Discussion by ngould** [duckdb/duckdb#5002] [General]
I just used `read_parquet` with a glob pattern, and got the error below:

![image](https://user-images.githubusercontent.com/3130631/195917554-ebe4a2ed-9e43-49ed-9b07-0c8a592ab4cc.png)

This struck me as a case that could be better handled by just returning an empty result set. Users may be iterating over some file naming scheme without knowing what files are actually present. In those cases (e.g. writing dbt transformations), it's often inconvenient to add exception handling logic outside of SQL.

Just a thought!

**Comment by hannes** [comment]
If nothing is found we don't know the result schema, so that may be problematic.

**Reply by ngould** [reply]
Fair point. Maybe there should be a way to (optionally) pass an expected result schema to `read_parquet()`? 

There is also the case of multiple comma separated pattern arguments, where duckdb may know the schema from an earlier pattern. Suppressing an exception seems possible there, but it’s not obvious that it would be good API design to do so by default.

**Comment by OzySky** [comment]
Perhaps a flag might be appropriate (like IGNORE_FILE_ERRORS) to return an empty result set ?

**Reply by OzySky** [reply]
Or skip that file for example (when just one file is missing)
